### ISSUE TITLE
Add CMake FetchContent support to triton client libraries

### ISSUE THREAD

**Issue by philipp-schmidt** [triton-inference-server/server#6759]
Is there a way to include tritonclient libraries in a CMake project via `FetchContent`? The following does not work properly.

```
include(FetchContent)
FetchContent_Declare(TritonClient
    GIT_REPOSITORY https://github.com/triton-inference-server/client
    GIT_TAG main
)
FetchContent_MakeAvailable(TritonClient)

# ...

target_link_libraries(test PRIVATE TritonClient::grpcclient rt m dl)
```

Including tritonclient via find_package will yield the target `TritonClient::grpcclient`, but via FetchContent the targets are not available.

**Comment by debermudez** [comment]
@Tabrizian Wouldnt this change need to go into the server repo?

**Comment by philipp-schmidt** [comment]
I have more info to share. The above code snippet misses setting the build flags for TritonClient to e.g. enable the grpc build. Doing so (via `SET(...)`) will give a target TritonClient::grpcclient, but it will fail to "transitively" find the target GRPC::xxx defined by triton. Tritonclient builds grpc itself so it should be available.

I can post more detailled info in a few days.

**Comment by debermudez** [comment]
@philipp-schmidt if you have the capacity, you can make a PR and fill out the [CLA](https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla). We can test locally and verify it.

**Comment by Tabrizian** [comment]
> @Tabrizian Wouldnt this change need to go into the server repo?

No, I think what @philipp-schmidt is proposing is to make changes to the client repo cmakes so that it works with `FetchContent` semantics.
